backend/internal/repository/starrocks_repository.go

```diff
@@ -309,3 +310,33 @@ func (r *StarrocksRepository) GetSequencing(seqId int) (*types.Sequencing, error
 	}
 	return &sequencing, err
 }
+
+func (r *StarrocksRepository) GetTermAutoComplete(termsTable string, input string) ([]types.AutoCompleteTerm, error) {
+	selectClause := fmt.Sprintf("id, name, REGEXP_REPLACE(name, '(?i)%s', '<strong>%s</strong>') AS highlighted_name", input, input)
+	whereClause := fmt.Sprintf("name REGEXP '(?i)%s'", input)
+	tx := r.db.Table(termsTable).Select(selectClause).Where(whereClause)
+	var terms []types.TermWithHighLight
+	err := tx.Find(&terms).Error
+	if err != nil {
+		if !errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, fmt.Errorf("error while fetching terms: %w", err)
+		} else {
+			return nil, nil
+		}
+	}
+
+	var output []types.AutoCompleteTerm
```

Comment:
can init the output array at the exact length `output := make([]*types.AutoCompleteTerm, len(terms))

Reply:
Done